### PR TITLE
Scan packages in parallel

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
Instead of scanning one package at a time, scan GOMAXPROCS at a time.

### CODE REVIEW

File: pkg/cli/scan.go [hunk 1]
```diff
@@ -205,19 +208,36 @@ wolfictl scan package1 package2 --remote
...
-					// Accumulate the list of failures to be returned at the end, but we still want to complete all scans
-					inputPathsFailingRequireZero = append(inputPathsFailingRequireZero, input)
-				}
+				g.Go(func() error {
+					isr, err := p.doScanCommandForSingleInput(cmd.Context(), input, advisoryDocumentIndices)
+					if err != nil {
+						return err
+					}
+
+					scans[i] = *isr
```
Comment (luhring) [inline pkg/cli/scan.go:225]: Are we sure this kind of slice access is thread-safe?

File: pkg/cli/scan.go [hunk 2]
```diff
@@ -205,19 +208,36 @@ wolfictl scan package1 package2 --remote
...
-				scans = append(scans, *isr)
+			for i, input := range inputs {
+				i, input := i, input
 
-				if p.requireZeroFindings && len(isr.Result.Findings) > 0 {
-					// Accumulate the list of failures to be returned at the end, but we still want to complete all scans
-					inputPathsFailingRequireZero = append(inputPathsFailingRequireZero, input)
-				}
+				g.Go(func() error {
+					isr, err := p.doScanCommandForSingleInput(cmd.Context(), input, advisoryDocumentIndices)
```
Comment (luhring) [inline pkg/cli/scan.go:220]: Actually, have we checked that everything called within this point is thread-safe? There's a ton happening here. Just curious!

File: pkg/cli/scan.go (hunk 1 above)
Comment (jonjohnsonjr) [reply pkg/cli/scan.go:225]: Yep! Appending isn't, which is why below on findings I use the lock to append to `inputPathsFailingRequireZero`.

File: pkg/cli/scan.go (hunk 2 above)
Comment (jonjohnsonjr) [reply pkg/cli/scan.go]: Sure. I've kept the actually scanning bits serial, but started fanning out the SBOM generation and also overlapped that with the vulnerability database loading.

**Comment by jonjohnsonjr** [comment]
Before: 
image

After: 
image

**Review by pdeslaur — APPROVED** [review]

**Pull request merged**